TCAT/Model/Route.swift

```diff
@@ -78,4 +95,52 @@ class Route: NSObject {
         self.travelDistance = travelDistance
         self.lastStopTime = lastStopTime
     }
+    
+    func directionJSON(json: [JSON]) -> [Direction] {
+        var directionArray = [Direction]()
+        for direction in json {
+            let directionType = direction["directionType"].stringValue
+            if directionType == "walk" {
+                directionArray.append(walkDirection(json: direction))
+            } else if directionType == "depart" {
+                directionArray.append(departDirection(json: direction))
+            } else {
+                directionArray.append(arriveDirection(json: direction))
+            }
+        }
+        return directionArray
+    }
+    
+    func walkDirection(json: JSON) -> WalkDirection {
+        let time = timeToDate(time: json["time"].stringValue)
+        let place = json["place"].stringValue
+        let location = CLLocation(latitude: json["location"][0].doubleValue, longitude: json["location"][1].doubleValue)
+        let travelDistance = json["travelDistance"].doubleValue
+        return WalkDirection(time: time, place: place, location: location, travelDistance: travelDistance)
+    }
+    func departDirection(json: JSON) -> DepartDirection {
+        let time = timeToDate(time: json["time"].stringValue)
+        let place = json["place"].stringValue
+        let location = CLLocation(latitude: json["location"][0].doubleValue, longitude: json["location"][1].doubleValue)
+        let routeNumber = json["routeNumber"].intValue
+        
+        let bound = Bound(rawValue: json["bound"].stringValue)
+        let stops = json["stops"].arrayObject as! [String]
+        let arrivalTime = timeToDate(time: json["arrivalTime"].stringValue)
+        return DepartDirection(time: time, place: place, location: location, routeNumber: routeNumber, bound: bound!, stops: stops, arrivalTime: arrivalTime)
+        
+    }
+    func arriveDirection(json: JSON) -> ArriveDirection {
+        let time = timeToDate(time: json["time"].stringValue)
+        let place = json["place"].stringValue
+        let location = CLLocation(latitude: json["location"][0].doubleValue, longitude: json["location"][1].doubleValue)
+        return ArriveDirection(time: time, place: place, location: location)
+        
+    }
+    
+    func timeToDate(time: String) -> Date {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "hh:mm a"
```

Comment:
there is a function in Time.swift that does this string(from time). Switch to it so we can stay consistent.